### PR TITLE
Modified the main menu and translation

### PR DESCRIPTION
**Pull request by Camilot55**
Hi, Giuseppe. I have everything prepared for the new version of the app. 
Summary of changes:
1. Modified main menu (ct_menu.сс)
2. Fixed the translation of ru.po. (cherrytree. pop did not fix it) 
3. In the files ct_menu. h and ct_pref_dlg_kb_shortcuts.cc fixed the names of the keys according to the keyboard. 
4. In the file ct_actions_edit.cc fixed the name of the dialog box for adding special characters. 
5. In the file ct_pref_dlg_theme.cc added a dialog for accepting or canceling default values. The user needs a choice! 
6. In the file ct_pref_dlg_text_n_code.cc added a hint for the help button.

### CODE REVIEW

File: src/ct/ct_menu.h [hunk 1]
```diff
@@ -56,11 +56,11 @@ class CtMenu
 
 public:
     const char*       None       = "";
-    const std::string KB_CONTROL = "<control>";
-    const std::string KB_SHIFT   = "<shift>";
-    const std::string KB_ALT     = "<alt>";
-    const std::string KB_META    = "<meta>";
-
```
Comment (giuspen) [inline src/ct/ct_menu.h:63]: This change is going to screw up all the existing custom keyboard shortcuts of the users in config.cfg
e.g.
```
[keyboard]
insert_special_char=<control><shift><alt><meta>S
```
Cannot accept this disruption, please revert it.

File: src/ct/ct_pref_dlg_kb_shortcuts.cc [hunk 2]
```diff
@@ -155,10 +155,10 @@ bool CtPrefDlg::edit_shortcut_dialog(std::string& shortcut)
     auto radiobutton_kb_none = Gtk::manage(new Gtk::RadioButton{_("No Keyboard Shortcut")});
     auto radiobutton_kb_shortcut = Gtk::manage(new Gtk::RadioButton{});
     radiobutton_kb_shortcut->join_group(*radiobutton_kb_none);
-    auto ctrl_toggle = Gtk::manage(new Gtk::ToggleButton{"control"});
-    auto shift_toggle = Gtk::manage(new Gtk::ToggleButton{"shift"});
-    auto alt_toggle = Gtk::manage(new Gtk::ToggleButton{"alt"});
-    auto meta_toggle = Gtk::manage(new Gtk::ToggleButton{"meta"});
```
Comment (giuspen) [inline src/ct/ct_pref_dlg_kb_shortcuts.cc:161]: here instead it is acceptable to change the labels of the ToggleButtons so you may leave this change

**Review by giuspen — CHANGES_REQUESTED** [review]

**Comment by giuspen** [comment]
Thanks Victor there is just one change and then I can merge.The change is because we don't want the users to lost the current custom keyboard shortcuts.

File: src/ct/ct_menu.h (hunk 1 above)
Comment (Camilot55) [reply src/ct/ct_menu.h:63]: undid changes in the ct_menu.h file

File: src/ct/ct_pref_dlg_kb_shortcuts.cc (hunk 2 above)
Comment (Camilot55) [reply src/ct/ct_pref_dlg_kb_shortcuts.cc:161]: Ok

**Comment by Camilot55** [comment]
Please add the following lines to the cherrytree.pot file. They participate in the translation
#: /home/giuseppe/git/cherrytree/src/ct/ct_pref_dlg.h:61
msgid "Are you sure to Reset to Default?"
msgstr ""

#: /home/giuseppe/git/cherrytree/src/ct/ct_imports.h:69
msgid "HTML Document"
msgstr ""

#: /home/giuseppe/git/cherrytree/src/ct/ct_imports.h:167
msgid "Plain Text Document"
msgstr ""

#: /home/giuseppe/git/cherrytree/src/ct/ct_imports.h:182
msgid "Markdown Document"
msgstr ""

#: /home/giuseppe/git/cherrytree/src/ct/ct_imports.h:216
msgid "Mempad Document"
msgstr ""

#: /home/giuseppe/git/cherrytree/src/ct/ct_imports.h:225
msgid "Treepad Document"
msgstr ""

#: /home/giuseppe/git/cherrytree/src/ct/ct_imports.h:235
msgid "Leo Document"
msgstr ""

**Comment by Camilot55** [comment]
There is a great library for syntax highlighting, see if you can use it in your application
[http://colorer.sourceforge.net/](url)

**Comment by Camilot55** [comment]
Sorry I clicked the wrong way, now I don't remember how to cancel this commit e770e19 to get everything back

**Comment by giuspen** [comment]
Thanks Victor. Note that the .pot is updated via the script https://github.com/giuspen/cherrytree/blob/master/po/i18n_update_pot.py and not manually. Similarly https://github.com/giuspen/cherrytree/blob/master/po/i18n_create_lang_file.py with argument kk should generate the Kazakh if you want to have a go.

**Comment by giuspen** [comment]
Don't worry I will do a manual merge and add the missing file back

**Comment by giuspen** [comment]
Merged manually, nice work thanks Victor

**Comment by Camilot55** [comment]
> Спасибо, Виктор. Обратите внимание, что .pot обновляется с помощью скрипта [https://github.com/giuspen/cherrytree/blob/master/po/i18n_update_pot.py,](https://github.com/giuspen/cherrytree/blob/master/po/i18n_update_pot.py) а не вручную. Аналогичным образом https://github.com/giuspen/cherrytree/blob/master/po/i18n_create_lang_file.py с аргументом kk должен генерировать казахский язык, если вы хотите попробовать.

Yes, thank you, I have long paid attention to these scripts. But here is a small problem I can't run them on Windows yet. For Linux, I have already tweaked a little.

**Comment by giuspen** [comment]
You should be able to run those scripts on Windows if you install msys2

**Comment by Camilot55** [comment]
I have msys2, but still something is missing.

**Comment by Camilot55** [comment]
======= RESTART: G:\msys64\home\Camilot\cherrytree\po\i18n_update_pot.py =======
Traceback (most recent call last):
  File "G:\msys64\home\Camilot\cherrytree\po\i18n_update_pot.py", line 19, in <module>
    subprocess.call(shell_cmd)
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\lib\subprocess.py", line 339, in call
    with Popen(*popenargs, **kwargs) as p:
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\lib\subprocess.py", line 800, in __init__
    restore_signals, start_new_session)
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\lib\subprocess.py", line 1207, in _execute_child
    startupinfo)
FileNotFoundError: [WinError 2] Не удается найти указанный файл

**Comment by Camilot55** [comment]
Hi Giuseppe, please use the following script to update cherrytree. pot
 Some lines are lost that can be translated.
Code i18n_update_pot.py
`#!/usr/bin/env python3

import os
import subprocess
import glob

APP_NAME = "cherrytree"
SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
SRC_CT_DIR = os.path.join(os.path.dirname(SCRIPT_DIR), "src", "ct")
CC_FILES = glob.glob(os.path.join(SRC_CT_DIR, "*.cc"))
H_FILES = glob.glob(os.path.join(SRC_CT_DIR, "*.h"))

shell_cmd = ["xgettext",
             "--language=C++",
             "--from-code=utf-8",
             "--keyword=_",
             "--output=" + os.path.join(SCRIPT_DIR, APP_NAME+".pot")
             ] + CC_FILES + H_FILES
subprocess.call(shell_cmd) `

**Comment by Camilot55** [comment]
Checked it out. Everything works here is the resulting new cherrytree. pot
[cherrytree.zip](https://github.com/giuspen/cherrytree/files/6207028/cherrytree.zip)

**Comment by Camilot55** [comment]
I have more suggestions to redo the app Settings menu. 
And you can also connect branches with plain text for spell checking.

**Comment by giuspen** [comment]
Victor you are meant to use the msys2 x64 console, that's why it is not working

**Comment by Camilot55** [comment]
OK. Thanks. I set everything up. Incorrectly wrote commands to run scripts. I'm new to python and C++. Basically, I wrote all the programs in Delphi. I want to learn C++. Help me? If there are any difficulties.

**Comment by giuspen** [comment]
http://www.cplusplus.com/doc/tutorial/ is a good place to start

**Pull request closed without merging**